src/platforms/hosted/cmsis_dap.c

```diff
@@ -468,7 +475,7 @@ static ssize_t dap_run_cmd_raw(const uint8_t *const request_data, const size_t r
 	const size_t result = (size_t)response;
 
 	DEBUG_WIRE("response: ");
-	for (size_t i = 0; i < result; i++)
+	for (size_t i = 0; i < response_length + 1; i++)
```

Comment:
result before was const size_t result = (size_t)response or 64.  Before the whole buffer was printed, now only the return code and the resulting data is printed.

Reply:
The whole point is we want the full used portion of the buffer printed as this is useful for debugging CMSIS-DAP issues and has been used multiple times since its introduction to solve problems in both adaptors and BMDA. `result = (size_t)response` does not result in 64 under all CMSIS-DAP implementations and results in the properly sized used portion of the buffer being used on CMSIS-DAP v2 adaptors and properly implemented v1 adaptors.

Even then, `response_length` is the total length of the response buffer, adding one is strictly an error and will result in reading a byte past the end - ASAN should have told you this.